### PR TITLE
Webdriverjs mocha wrapper for 'it' should support the 'done' callback

### PR DESCRIPTION
**Pull request by tullmann**
This pull request has three changes in it, one fixes some doc typos, one adds a test for the handling of the `done` callback on an `it`, and one patch fixes the bug.

I believe this it the issue https://code.google.com/p/selenium/issues/detail?id=7277.  The fix in https://code.google.com/p/selenium/source/detail?r=88f6680bf0b5 does not, as far as I can see (or experience), do the right thing.

I wasn't able to test this through "./go" (The documentation step blows up with 604 warnings?), so I only ran the mocha tests manually.

### CODE REVIEW

**Comment by andreastt** [comment]
Thanks for the patch!  Superficially this looks good to me, but @jleyba is more of an expert in this area and should have a look.

File: javascript/node/selenium-webdriver/testing/index.js [hunk 1]
```diff
@@ -90,39 +90,71 @@ function seal(fn) {
...
+        // ignored.
+        testFn(function testFnDoneCallback(err) {
+          if (err) {
+            testFnDone.reject(err);
+          } else {
+            testFnDone.fulfill();
+          }
+        });
+      }
+      else {
```
Comment (jleyba) [inline javascript/node/selenium-webdriver/testing/index.js]: style nit: `} else {`

File: javascript/node/selenium-webdriver/test/testing/index_test.js [hunk 2]
```diff
@@ -105,3 +105,59 @@ describe('Mocha Integration', function() {
...
+   // This it blocks until 'done' is invoked
+   it('vanilla delayed', function it_vanillaDelayed(done) {
+      setTimeout(function delayed_vanillaTimeout() {
+         waited = true;
+         done();
+      }, DELAY);
+   });
+
+   it('vanilla must have waited', function it_vanillaMustHaveWaited() {
+      assert.strictEqual(waited, true);
```
Comment (jleyba) [inline javascript/node/selenium-webdriver/test/testing/index_test.js]: Tests should not depend on ordering.  Use

    beforeEach(function() { waited = false; });
    afterEach(function() { assert.strictEqual(waited, true); });

File: javascript/node/selenium-webdriver/test/testing/index_test.js [hunk 3]
```diff
@@ -105,3 +105,59 @@ describe('Mocha Integration', function() {
...
+describe('Mocha async "done" support', function() {
+   this.timeout(2*1000);
+
+   var waited = false;
+   var DELAY = 100; // ms enough to notice
+
+   // --- First, vanilla mocha "it" should support the "done" callback correctly.
+
+   // This it blocks until 'done' is invoked
+   it('vanilla delayed', function it_vanillaDelayed(done) {
```
Comment (jleyba) [inline javascript/node/selenium-webdriver/test/testing/index_test.js]: style nit: use camelCase, not snake_case

**Comment by tullmann** [comment]
Thanks for the feedback, I've addressed all jleyba's comments.

Let me know if you'd like me to fix anything else or rebase this or reorder the commits or whatever.  I've pushed another commit with the requested fixes into this PR.

**Comment by jleyba** [comment]
Have you signed the CLA? http://goo.gl/qC50R

**Comment by tullmann** [comment]
Yep, I filled out the CLA form a couple days ago.

**Comment by jleyba** [comment]
Please rebase and squash your commits. Also make sure tests pass - I know the ./go command is broken right now. Easiest to just `cd javascript/node/selenium-webdriver && npm test`

**Comment by tullmann** [comment]
I got the tests working via `npm` (good thing too, as a couple were indirectly broken by this change).  I've pushed a change here that fixes the tests (basically can't pass "httpserver.start" directly to "before", as it has an optional parameter, which before interprets as an async callback parameter).  I'll will rebase, squash everything and push a new pull request.

I ran the tests with Chrome and Firefox.  Two Chrome-based tests failed setting the "loggingPrefs" of Type.BROWSER.  They also failed in a clean tree, too, though, so I ignored them.

>  1547 passing (2m)
>  3 pending
>  2 failing

**Comment by tullmann** [comment]
Replaced by #318

**Pull request closed without merging**